client/src/pages/Checkout.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from "react";
+import { usePageLoaded } from "../hooks";
+import {
+  CardElement,
+  Elements,
+  Stripe,
+  useStripe,
+  useElements,
+} from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import { createPaymentIntent } from "../services";
+
+const showCart = (cart) => {
+  // only show most recent
+  const item = cart[0];
+  return (
+    <table>
+      <tbody>
+        <tr className="">
+          <td>Item</td>
+          <td>Quantity</td>
+          <td>Unit Cost</td>
+          <td>Total Cost</td>
+        </tr>
+        <tr className="">
+          <td className="">{item.name}</td>
+          <td className="">{item.quantity}</td>
+          <td className="">${item.unitCost}</td>
+          <td className="">${item.unitCost * item.quantity}</td>
+        </tr>
+      </tbody>
+    </table>
+  );
+};
+
+const stripePromise = loadStripe("key");
+const Checkout = ({ state, dispatch }) => {
+  usePageLoaded(dispatch);
```

Comment:
I actually don't really see the purpose of this hook, is it really necessary at this time to have it? It's just to be there when the page is loaded?

Reply:
In this case I wanted to lock input once the user submits the cart on `/balance` and unlock once the Checkout component mounts. I assumed this kind of control might be useful in other areas of the app where data flows through multiple pages and it was a pretty trivial to put in a hook.